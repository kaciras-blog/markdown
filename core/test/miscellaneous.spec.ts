import { expect, it } from "vitest";
import MarkdownIt from "markdown-it/lib";
import { Anchor, Classify, Footnote, UGC } from "../src/index.ts";

it("should set the attribute", () => {
	const markdownIt = new MarkdownIt();
	markdownIt.use(UGC);

	const html = markdownIt.render("[test](http://example.com)");
	expect(html.trimEnd()).toBe('<p><a href="http://example.com" rel="ugc,nofollow">test</a></p>');
});

it("should add class to inlined code block", () => {
	const markdownIt = new MarkdownIt();
	markdownIt.use(Classify);

	expect(markdownIt.render("`foobar`")).toMatchSnapshot();
});

it("should add anchor to titles", () => {
	const markdownIt = new MarkdownIt();
	markdownIt.use(Anchor);

	expect(markdownIt.render("# foobar")).toMatchSnapshot();
});

it("should support footnote common", () => {
	const markdownIt = new MarkdownIt();
	markdownIt.use(Footnote);

	const html = markdownIt.render("test[^1]\n\n[^1]: foobar\n\nafter");
	expect(html).toMatchSnapshot();
});
