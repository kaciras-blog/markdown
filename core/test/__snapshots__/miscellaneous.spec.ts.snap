// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should add anchor to titles 1`] = `
"<h1 id="foobar" tabindex="-1">foobar <a class="anchor-link" href="#foobar">#</a></h1>
"
`;

exports[`should add class to inlined code block 1`] = `
"<p><code class="inline-code">foobar</code></p>
"
`;

exports[`should add docId to anchor link 1`] = `
"<h1 id="8964-foobar" tabindex="-1">foobar <a class="anchor-link" href="#8964-foobar">#</a></h1>
"
`;

exports[`should support footnote common 1`] = `
"<p>test<sup class="footnote-ref"><a href="#fn1" id="fnref1">[1]</a></sup></p>
<p>after</p>
<h2 class='footnotes'></h2><ol class='footnotes-list'><li id="fn1" class="footnote-item"><p>foobar <a href="#fnref1" class="footnote-backref">↩︎</a></p>
</li>
</ol>"
`;

exports[`should wrap table with a div 1`] = `
"<div class="table-view"><table><thead>
<tr>
<th>head</th>
</tr>
</thead>
<tbody>
<tr>
<td>cell</td>
</tr>
</tbody>
</table></div>"
`;
